### DISCUSSION TITLE
Warum wird mit "L" eingelesener mehrzeiliger Befehl mit "Einfügen" falsch eingefügt?

### DISCUSSION THREAD

**Discussion by helmutkilian** [TobiasPSP/OReilly#9] [Fragen] [answered]
Liest man mit dem Befehl 
`L 1.3 `
das Listing 1.3 in die Zwischenablage ein und fügt den in 1.3 enthaltenen **mehrzeiligen Befehl** dann mit "Strg+V" in die Konsole ein, so funktioniert alles wie erwartet.

Benutzt man zum Einfügen in die Konsole statt "Strg+V" jedoch "Einfügen" aus dem Konsolen-Kontextmenü (rechte Maustaste), so werden die Zeilen aus Listing 1.3 in **umgekehrter Reihenfolge** in die Konsole eingefügt!?? Dann kann man den Befehl natürlich so nicht ausführen. Ich kann mir diese Umkehrung der Zeilenfolge nicht erklären.

**Comment by helmutkilian** [comment, marked as answer]
Ich habe inzwischen selbst nachgesehen, wo dieser merkwürdige Effekt herkommt (da ich - zugegeben hier unnötigerweise - gewöhnlich den "Quick-Edit-Modus" mit Maus-Rechtsklick nutze) und habe die Ursache auch gefunden. In dem Ladeskript ist es die Zeile mit der Übergabe ans Clipboard:
    `$code.Trim() | Set-ClipBoard`
`Set-ClipBoard` erwartet laut Hilfe ein String[] für die Eingabe-Zeilen. Von `$code` bekommt es aber einen Gesamt-String und zerlegt diesen nur dann automatisch in Zeilen, wenn der Windows-Zeilentrenner CRLF drinsteckt. In `$code` steckt aber nur LF drin.

Ich benutze nun für die Definition von "L"
`(Invoke-RestMethod -Uri https://tinyurl.com/codeAusBuch).replace('$code.Trim() | Set-ClipBoard', '$code.Trim().Split("<backquote>n") | Set-ClipBoard') | New-Item -Path function: -Name L -Force | Out-Null`

(Einzelnes backquote lässt sich hier in der Code-Auszeichnung offenbar nicht darstellen, weshalb ich hier `<backquote>` statt ` schreibe.)

Von nun an ist es egal, wie man nach Ausführung von `L x.y` den Inhalt der Zwischenablage in die Konsole einfügt.

**Comment by TobiasPSP** [comment]
Vielen Dank Helmut für diese gute Lösung. Wie ich schon bei Deinem anderen Beitrag angemerkt habe, sind mir die Benachrichtigungen für Deine Posts irgendwie durchgerutscht. Daher der etwas sehr späte Kommentar. Mea culpa.
